src/shapes/polyline.class.ts

```diff
@@ -235,19 +304,25 @@ import { makeBoundingBoxFromPoints } from '../util/misc/boundingBoxFromPoints';
    * @param {Object} [options] Options object
    */
   fabric.Polyline.fromElementGenerator = function (_class) {
-    return function (element, callback, options) {
+    return function (element, callback, options = {}) {
       if (!element) {
         return callback(null);
       }
-      options || (options = {});
-
-      var points = fabric.parsePointsAttribute(element.getAttribute('points')),
-        parsedAttributes = fabric.parseAttributes(
+      const points = parsePointsAttribute(element.getAttribute('points')),
+        parsedAttributes = parseAttributes(
           element,
           fabric[_class].ATTRIBUTE_NAMES
         );
-      parsedAttributes.fromSVG = true;
-      callback(new fabric[_class](points, extend(parsedAttributes, options)));
+      callback(
+        new fabric[_class](points, {
+          ...parsedAttributes,
+          ...options,
+          // we pass undefined to instruct the constructor to position the object using the bbox
+          left: undefined,
+          top: undefined,
```

Comment:
left and top can just be in parsedAttributes. can we just do:
```js
const { left, top, allOtherAttributes } = parseAttributes(
          element,
          fabric[_class].ATTRIBUTE_NAMES
        );
```

and avoid setting back to undefined?

Also why this change? i ll read the description, if you know is not explained there can you elaborate?

Reply:
good point

Reply:
actually not.
since object has default values of `{ top: 0, left: 0}` we must pass undefined

Reply:
This is not a real change.
`_setPositionDimensions` used this logic but was written differently depending on `fromSVG` instead.

Reply:
Once `exactBoundingBox=true` by default `fromSVG` can be removed as well. That's why I changed it.

Reply:
yes this is hacky this is the exact part i wanted to give a second look at.

Reply:
Suggestions?

Reply:
i m not entirely sure. let me craft some weird SVG on purpose.
i do think that if you import from SVG we have to do something different and solve it in the svg import function.

Reply:
Agreed

Reply:
I have an idea
Is this the only use case to set position in the constructor?
If so we move that logic to from svg aftet initialization

Reply:
Could we use `pathOffset` for that?
```js
setPositionByOrigin(poly.pathOffset, 'center', 'center')
```

Reply:
Do you mind we do that in the follow up?

Reply:
Then we add dedicated tests for svg cases as well

Reply:
I looked at this again I saw my comments are wrong as well.
No need to pass undefined as you said since the code checks if passed options have left, top **not** default value.

Too many open PRs so they leak in my mind to each other. I did something in a PR and then I work on 3 more and I start assuming wrong things.